tests/unit/test_main.py

```
@@ -553,3 +569,246 @@ def test_integration_complete_workflow():
+    mock_fallback_result.provider_name = 'ollama'
+    mock_fallback_result.had_fallback = True
+    mock_fallback_result.failed_providers = [('openai', 'API key invalid')]
+
+    with patch('sys.argv', test_args):
+        with patch('hai_sh.__main__.init_hai_directory', return_value=(True, None)):
+            with patch('hai_sh.__main__.load_config', return_value=mock_config):
+                with patch('hai_sh.__main__.get_available_provider', return_value=mock_fallback_result):
+                    with patch('hai_sh.__main__.generate_with_retry', return_value=mock_response):
+                        with patch('hai_sh.__main__.get_user_confirmation', return_value=True):
+                            with patch('hai_sh.__main__.execute_command', return_value=mock_result):
+                                with patch('sys.stdout', new_callable=io.StringIO) as mock_stdout:
```

Ruff (0.14.10)

Local variable `mock_stdout` is assigned to but never used

Remove assignment to unused variable `mock_stdout`

(F841)